gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,498 @@
+  }
+
+  private PublisherApi(ServiceApiSettings settings) throws IOException {
+    ServiceApiSettings internalSettings = ApiUtils.settingsWithChannels(settings,
+        SERVICE_ADDRESS, DEFAULT_SERVICE_PORT, ALL_SCOPES);
+    this.settings = internalSettings;
+    this.channel = internalSettings.getChannel();
+  }
+
+  // ==============================
+  // Resource Name Helper Functions
+  // ==============================
+
+  public static final String createProject(String project) {
+    return PROJECT_PATH_TEMPLATE.instantiate(
+        "project", project);
+  }
+
+  public static final String createTopic(String project, String topic) {
+    return TOPIC_PATH_TEMPLATE.instantiate(
+        "project", project,"topic", topic);
+  }
+
+
+  // ========
+  // Getters
+  // ========
+
+  public Channel getChannel() {
+    return channel;
+  }
+
+
+  // =============
+  // Service Calls
+  // =============
+
+  // ----- createTopic -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   *
+   * @param name The name of the topic. It must have the format
+   * `"projects/{project}/topics/{topic}"`. `{topic}` must start with a letter,
+   * and contain only letters (`[A-Za-z]`), numbers (`[0-9]`), dashes (`-`),
+   * underscores (`_`), periods (`.`), tildes (`~`), plus (`+`) or percent
+   * signs (`%`). It must be between 3 and 255 characters in length, and it
+   * must not start with `"goog"`.
+   */
+  public Topic createTopic(String name) {
+    Topic request =
+        Topic.newBuilder()
+        .setName(name)
+        .build();
+
+    return createTopic(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public Topic createTopic(Topic request) {
+    return createTopicCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates the given topic with the given name.
+   */
+  public ApiCallable<Topic, Topic> createTopicCallable() {
+    return ApiUtils.prepareIdempotentCallable(CREATE_TOPIC, settings).bind(channel);
+  }
+
+  // ----- publish -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Adds one or more messages to the topic. Returns NOT_FOUND if the topic does
+   * not exist. The message payload must not be empty; it must contain either a
+   * non-empty data field, or at least one attribute.
+   *
+   * @param topic The messages in the request will be published on this topic.
+   * @param messages The messages to publish.
+   */
+  public PublishResponse publish(String topic, List<PubsubMessage> messages) {
```

Comment:
Is issue #384 depends on changing pubsub gRPC API to include stream operations?
If so, does the gax layer already support gRPC stream operations?